lib/widgets/topic_list.dart

```diff
@@ -0,0 +1,301 @@
+      ),
+      body: TopicListPageBody(streamId: streamId),
+    ));
+  }
+}
+
+class _TopicListAppBarTitle extends StatelessWidget {
+  const _TopicListAppBarTitle({
+    required this.streamId,
+    required this.willCenterTitle,
+  });
+
+  final int streamId;
+  final bool willCenterTitle;
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final stream = store.streams[streamId];
+    final subscription = store.subscriptions[streamId];
+    final designVariables = DesignVariables.of(context);
+
+    final iconData = subscription != null ? iconDataForStream(subscription) : null;
+    final alignment = willCenterTitle
+      ? Alignment.center
+      : AlignmentDirectional.centerStart;
+
+    return SizedBox(
+      width: double.infinity,
+      child: Align(alignment: alignment,
+        child: Row(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            Icon(size: 16, iconData),
+            const SizedBox(width: 4),
+            Flexible(child: Row(
+              children: [
+                Text(
+                  stream?.name ?? zulipLocalizations.unknownChannelName,
+                  style: recipientHeaderTextStyle(context),
+                ),
+                const SizedBox(width: 10),
+                Icon(ZulipIcons.chevron_down, size: 10, color: designVariables.icon),
+              ],
+            )),
+          ]))
+    );
+  }
+}
+
+class TopicListPageBody extends StatefulWidget {
+  const TopicListPageBody({super.key, required this.streamId});
+
+  final int streamId;
+
+  @override
+  State<TopicListPageBody> createState() => _TopicListPageBodyState();
+}
+
+class _TopicListPageBodyState extends State<TopicListPageBody>
+    with PerAccountStoreAwareStateMixin<TopicListPageBody> {
+  TopicListView? _model;
+  Unreads? _unreadsModel;
+
+  @override
+  void onNewStore() {
+    _unreadsModel?.removeListener(_modelChanged);
+    final store = PerAccountStoreWidget.of(context);
+    _unreadsModel = store.unreads..addListener(_modelChanged);
+
+    _model?.dispose();
+    _model = TopicListView(store: store, streamId: widget.streamId);
+    _model!.addListener(_modelChanged);
+    _fetchTopics();
+  }
+
+  Future<void> _fetchTopics() async {
+    try {
+      await _model!.fetchTopics();
+      if (_model!.hasError && mounted) {
+        final zulipLocalizations = ZulipLocalizations.of(context);
+        showErrorDialog(
+          context: context,
+          title: zulipLocalizations.errorFetchingTopics,
+          message: _model!.errorMessage);
+      }
+    } catch (e) {
+      if (!mounted) return;
+      final zulipLocalizations = ZulipLocalizations.of(context);
+      showErrorDialog(
+        context: context,
+        title: zulipLocalizations.errorFetchingTopics,
+        message: e.toString());
+    }
+  }
```

Comment:
I think with the way how `fetchTopics` is used right now, having a separate class for the data does not feel like a helpful abstraction, since it adds a layer of indirection as a trade-off.

Right now, we only fetch topics `onNewStore`, and there is no retry logic yet. For error handling, The `_fetchTopics` helper does not really care if `errorMessage` lives on as a part of the model. This also requires as to manage an additional listener of a `ChangeNotifier`.

It would probably be cleaner to inline the `getStreamTopics` here, and only consider extracting a model when there is more complexity that warrants a thicker data model. The only mutable states needed are `isLoading` and `topics`.